src/gui/folderwatcher_mac.cpp

```diff
@@ -105,9 +104,29 @@ void FolderWatcherPrivate::startWatching()
     FSEventStreamStart(_stream);
 }
 
+void FolderWatcherPrivate::checkCoalescedPaths(QStringList &paths)
+{
+    QStringList coalescedPaths;
+    for (const auto &eventPath : qAsConst(paths)) {
+        if (QDir(eventPath).exists()) {
+            QDirIterator it(eventPath, QDir::AllDirs | QDir::NoDotAndDotDot, QDirIterator::Subdirectories);
+            while (it.hasNext()) {
+                auto subfolder = it.next();
+                if (!paths.contains(subfolder)) {
+                    coalescedPaths.append(subfolder);
+                }
+            }
+        }
+    }
+    paths.append(coalescedPaths);
+
+    doNotifyParent(paths);
+}
+
 void FolderWatcherPrivate::doNotifyParent(const QStringList &paths)
 {
-    _parent->changeDetected(paths);
+
+    _parent->changeDetected(qAsConst(paths));
```

Comment:
qAsConst is unnecessary here. This is already const even.

Now what I wonder is why do it like you did. Why not have a function which has the responsibility of completing the path list like you did and returning it. Then call this from doNotifyParent?
I find the current chaining with doNotifyParent going one level deeper in the call stack less readable overall.
Or am I missing something?

Reply:
1) You're right, this is a remnant of a previous but reverted removal of the const in doNotifyParent arguments.. Will remove.

2) Also good point, for me it was the most intuitive way, but intuition isn't necessarily reasonable. I get your point with the call stack.